src/cell.hpp

```diff
@@ -99,6 +99,11 @@ class cell {
          }
      }
 
+    /// Return the kind of cell, used for grouping into cell_groups
+    cell_kind const get_cell_kind() const  {
```

Comment:
Renaming `cell` to `mc_cell` (or even `mc_cell_description`) I think is sound, and other sorts of cell or fake cell will need different description objects. But is there anything really in common across them? I don't think we need an abstract base class.

When a recipe says it has a multi-compartment cell, we can ask it for the corresponding `mc_cell` description, either via a dedicated method that can return e.g. an `optional` value, or via a common method that returns a variant- or type-erased type.

I think this is actually a good use case for `std::any` (or more precisely, our implementation of it, given that that is C++17 and/or Boost).